client/QuickStart/Program.cs

```diff
@@ -0,0 +1,767 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Security.Cryptography.X509Certificates;
+using Kusto.Data;
+using Kusto.Data.Common;
+using Kusto.Data.Net.Client;
+using Kusto.Ingest;
+using Newtonsoft.Json;
+using Kusto.Cloud.Platform.Data;
+using ShellProgressBar;
+
+namespace QuickStart
+{
+    /// <summary>
+    /// ConfigJson object -  represents a cluster and DataBase connection configuration file.
+    /// </summary>
+    public class ConfigJson
+    {
+        public bool? UseExistingTable = null;
+        public string DatabaseName = null;
+        public string TableName = null;
+        public string TableSchema = null;
+        public string KustoUri = null;
+        public string IngestUri = null;
+        public string CertificatePath = null;
+        public string CertificatePassword = null;
+        public string ApplicationId = null;
+        public string TenantId = null;
+        public List<Dictionary<string, string>> Data = null;
+        public bool? AlterTable = null;
+        public bool? QueryData = null;
+        public bool? IngestData = null;
+    }
+
+
+    /// <summary>
+    /// The quick start application is a self-contained and runnable example script that demonstrates authenticating,
+    /// connecting to, administering, ingesting data into and querying Azure Data Explorer using the azure-kusto C# SDK.
+    /// You can use it as a baseline to write your own first kusto client application, altering the code as you go,
+    /// or copy code sections out of it into your app.
+    /// Tip: The app includes comments with tips on recommendations, coding best practices, links to reference materials
+    /// and recommended TO DO changes when adapting the code to your needs.
+    /// </summary>
+    public static class KustoSampleApp
+    {
+        //########################### Constants Start ###########################// 
+
+        // TODO (config - optional): Change the authentication method from "User Prompt" to any of the other options
+        //  Some of the auth modes require additional environment variables to be set in order to work
+        // (see usage in generate_connection_string below).
+        //  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+        private const string
```

Comment:
This line break is ugly. as yochai said - 160 chars per line is good, if you're having trouble defining it rider talk to me